src/Native/System.Native/pal_readdir.h

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+
+#include <dirent.h>
```

Comment:
Replace with struct DIR; forward declaration. I want to keep the headers needed for the implementation in the cpp.